src/ploomber_core/telemetry/telemetry.py

```diff
@@ -682,7 +691,19 @@ def wrapper(*args, **kwargs):
 
                 try:
                     if payload:
-                        result = func(_payload, *args, **kwargs)
+                        if is_method:
+                            """
+                            If the method is defined in Class, the original parameter:
+                            def method(self, parameter1, parameter2,...)
+                            We will inject the payload parameter right after the self
+                            To become:
+                            def method(self, payload, parameter1, parameter2,...)
+                            """
+                            injected_args = list(args)
```

Comment:
do we need the `list()`? what's the type of args?

Reply:
Per my test `args` is `type: <class 'tuple'>`
I need to insert `payload` right after the first arg `self`, so I created the ordered list to do the operation